### PR TITLE
Enhanced vulkan triangle app

### PR DESCRIPTION
**Pull request by hjred**
# Summary of the PR
The current Vulkan triangle sample code has some problems:
1. The Vulkan validation layer issues some warnings/errors
2. Resizing crashes the app

This PR will:
- fix the warnings/errors pointed out in the validation layers
- make window resizing work
- continously update window when resizing, as this removes artifacts when resizing

# Further Comments
The changes more or less follow the Vulkan Tutorial which is already the base for this code until Swapchain recreation:
https://vulkan-tutorial.com/Drawing_a_triangle/Swap_chain_recreation

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=475) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=475) before we can accept your contribution.<br/><br/>:x: hjred [sign now](https://cla.dotnetfoundation.org/dotnet/Silk.NET?pullRequest=475)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/Silk.NET?pullRequest=475) it.</sub>

**Review by Perksey — CHANGES_REQUESTED** [review]
This is awesome, thanks so much! Just a few tiny nitpicks, most of this was superb...

File: src/Lab/VulkanTriangle/HelloTriangleApplication.cs [hunk 1]
```diff
@@ -182,30 +217,39 @@ private unsafe void Cleanup()
...
+                _debugUtils.DestroyDebugUtilsMessenger(_instance, _debugMessenger, null);
+            }
+
+            _vkSurface.DestroySurface(_instance, _surface, null);
+            _vk.DestroyInstance(_instance, null);
+        }
+
+        private unsafe void CleanupSwapchain()
+        {
+            foreach (Framebuffer framebuffer in _swapchainFramebuffers)
```
[the remark this answers is not in the document]
Comment (HurricanKai) [reply src/Lab/VulkanTriangle/HelloTriangleApplication.cs:233]: 90% sure that's not necessary

File: src/Lab/VulkanTriangle/HelloTriangleApplication.cs [hunk 2]
```diff
@@ -182,30 +217,39 @@ private unsafe void Cleanup()
...
+            {
+                _vk.FreeCommandBuffers(_device, _commandPool, (uint) _commandBuffers.Length, buffers);
+            }
+
             _vk.DestroyPipeline(_device, _graphicsPipeline, null);
             _vk.DestroyPipelineLayout(_device, _pipelineLayout, null);
             _vk.DestroyRenderPass(_device, _renderPass, null);
 
-            foreach (var imageView in _swapchainImageViews)
+            foreach (ImageView imageView in _swapchainImageViews)
```
[the remark this answers is not in the document]
Comment (hjred) [reply src/Lab/VulkanTriangle/HelloTriangleApplication.cs:247]: The code for this is directly translated from the vulkan tutorial. The existing code did come from there too.

All of the changes are based on the chapter https://vulkan-tutorial.com/Drawing_a_triangle/Swap_chain_recreation
or more specifically the code file for this at https://github.com/Overv/VulkanTutorial/blob/master/code/16_swap_chain_recreation.cpp

I only tested things out and translated it to Silk.NET.
What I understand so far from Vulkan is that if the framebuffer size changes half of the Vulkan objects need to be destroyed and recreated as all of them are in some way or the other dependent on the framebuffer size/size of the resulting image.

[the remark this answers is not in the document]
Comment (Perksey) [reply src/Lab/VulkanTriangle/HelloTriangleApplication.cs:247]: Oh yeah I get that. My point is for the `foreach` loop I don't think you need to specify the type ;)

i.e. you can change `ImageView` to `var` per our code style and the C# compiler will still resolve it to the correct type (`ImageView`)

**Review by Perksey — APPROVED** [review]

File: src/Lab/VulkanTriangle/HelloTriangleApplication.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hjred) [reply src/Lab/VulkanTriangle/HelloTriangleApplication.cs]: :sweat_smile: sorry, it's late today.
No special reason, it's more a code style I prefer to have the type on the left and leave it out on the right if possible, so it is a little bit clearer to me to reason about the types without having to rely on IntelliSense.

Commited a fix

**Pull request merged**